resources/views/dimmers.blade.php

```diff
@@ -1,7 +1,17 @@
 @php
 $dimmers = \Arrilot\Widgets\Facade::group('voyager::dimmers');
-$count = $dimmers->count();
 
+// Loop over the defined dashboard widgets. Get the related model and remove the
+// widget from the group if the current user hasn't the permission to read it.
+foreach (config('voyager.dashboard.widgets') as $widgetClass) {
+    $relatedModel = app($widgetClass)->getRelatedModel();
+
+    if (! Auth::user()->can('read', $relatedModel)) {
```

Comment:
How about making the permission check being inside the dimmer? Making the dimmer have the logic. Makes it easier to overwrite for specific cases.

Reply:
I already tried that, but the problem is. We can't display the widget by condition. When adding a simple @can check to dimmer.blade.php the grid will look like: `dimmer 1 -- empty collumn -- dimmer 3` for example. I also tried to use the ->wrap() method to only add a div if the widget is accessible, but this method isn't passing the widget class throught.

Reply:
I don't see the problem, now you have

```
$relatedModel = app($widgetClass)->getRelatedModel();
```

Then your could have something like this instead:
```
if (!app($widgetClass)->accessable()) {
    $dimmers->removeByName($widgetClass);
}
```

Reply:
Ahh, I misunderstood you. Yes, that approach looks much cleaner! Nice!

Now I have this:

```php
/**
 * Determine if the widget is accessible.
 *
 * @return bool
*/
public function isAccessible()
{
    return \Auth::user()->can('browse', Voyager::model('Page'));
}
```

And in `dimmers.blade.php`:

```php
$dimmers = \Arrilot\Widgets\Facade::group('voyager::dimmers');

// Remove all the inaccessible widgets from the `voyager::dimmers` group.
foreach (config('voyager.dashboard.widgets', []) as $widgetClass) {
    if (!app($widgetClass)->isAccessible()) {
        $dimmers->removeByName($widgetClass);
    }
}
```

Alternatively, we could check the existance of the `isAccessible` method before calling it, but this could lead to unexpected behaviour.